nomos-services/sdp/src/lib.rs

```diff
@@ -133,17 +153,163 @@ where
                     }
                 }
                 SdpMessage::PostActivity { metadata, .. } => {
-                    tracing::debug!("todo: implement post activity {:?}", metadata);
+                    self.handle_post_activity(metadata, &wallet_adapter, &mempool_adapter)
+                        .await;
+                }
+                SdpMessage::PostDeclaration {
+                    declaration,
+                    reply_channel,
+                } => {
+                    self.handle_post_declaration(
+                        declaration,
+                        &wallet_adapter,
+                        &mempool_adapter,
+                        reply_channel,
+                    )
+                    .await;
+                }
+                SdpMessage::PostWithdrawal { declaration_id } => {
+                    self.handle_post_withdrawal(declaration_id, &wallet_adapter, &mempool_adapter)
+                        .await;
                 }
-                SdpMessage::PostDeclaration { .. } => todo!("implement post declaration"),
-                SdpMessage::PostWithdrawal { .. } => todo!("implement post withdrawal"),
             }
         }
 
         Ok(())
     }
 }
 
+impl<Backend, RuntimeServiceId> SdpService<Backend, RuntimeServiceId>
+where
+    Backend: SdpBackend + Send + Sync + 'static,
+    RuntimeServiceId: AsServiceId<Self> + Clone + Display + Send + Sync + 'static,
+{
+    async fn handle_post_declaration(
+        &self,
+        declaration: Box<DeclarationMessage>,
+        wallet_adapter: &MockWalletAdapter,
+        mempool_adapter: &MockMempoolAdapter,
+        reply_channel: oneshot::Sender<Result<DeclarationId, DynError>>,
+    ) {
+        let tx_builder = MantleTxBuilder::new();
+
+        let signed_tx = match wallet_adapter.declare_tx(tx_builder, declaration.clone()) {
+            Ok(tx) => tx,
+            Err(e) => {
+                tracing::error!("Failed to create declaration transaction: {:?}", e);
+                return;
+            }
+        };
```

Comment:
```suggestion
        let Ok(signed_tx) = wallet_adapter.declare_tx(tx_builder, declaration.clone()) else {
                tracing::error!("Failed to create declaration transaction: {:?}", e);
                return;
        };
```

Reply:
We would swallow the error in this case then, right?